openmc/mesh.py

```diff
@@ -2167,6 +2171,84 @@ def from_hdf5(cls, group: h5py.Group, mesh_id: int, name: str):
 
         return mesh
 
+    @classmethod
+    def from_domain(
+        cls,
+        domain: 'openmc.Cell' | 'openmc.Region' | 'openmc.Universe' | 'openmc.Geometry',
+        dimension: Sequence[int] = (10, 10, 10),
+        mesh_id: int | None = None,
+        phi_grid_bounds: Sequence[float] = (0.0, 2*pi),
+        theta_grid_bounds: Sequence[float] = (0.0, pi),
+        name: str = ''
+    ):
+        """Creates a regular SphericalMesh from an existing openmc domain.
+
+        Parameters
+        ----------
+        domain : openmc.Cell or openmc.Region or openmc.Universe or openmc.Geometry
+            The object passed in will be used as a template for this mesh. The
+            bounding box of the property of the object passed will be used to
+            set the r_grid, theta_grid, phi_grid ranges.
+        dimension : Iterable of int
+            The number of equally spaced mesh cells in each direction (r_grid,
+            theta_grid, phi_grid)
+        mesh_id : int
+            Unique identifier for the mesh
+        phi_grid_bounds : numpy.ndarray
+            Mesh bounds points along the phi-axis in radians. The default value
+            is (0, 2π), i.e., the full phi range.
+        theta_grid_bounds : numpy.ndarray
+            Mesh bounds points along the theta-axis in radians. The default value
+            is (0, π), i.e., the full theta range.
+        name : str
+            Name of the mesh
+
+        Returns
+        -------
+        openmc.SphericalMesh
+            SphericalMesh instance
+
+        """
+        cv.check_type(
+            "domain",
+            domain,
+            (openmc.Cell, openmc.Region, openmc.Universe, openmc.Geometry),
+        )
+
+        # loaded once to avoid recalculating bounding box
+        cached_bb = domain.bounding_box
+
+        outer_radius = 0.5 * max(cached_bb.width)
+
+        r_grid = np.linspace(
+            0,
+            outer_radius,
+            num=dimension[0]+1
+        )
+        theta_grid = np.linspace(
+            theta_grid_bounds[0],
+            theta_grid_bounds[1],
+            num=dimension[1]+1
+        )
```

Comment:
Wouldn't you want the default behavior for these bounds to be uniform in cos(theta)?

Reply:
Right. The cosine of theta could be better for certain applications, but I think I'd leave that to the user to replace the grid in those situations really. Same could be said about applying a power law to the the `r_grid` spacing if we're looking to make volumes equal or something.

Let me know if you feel strongly otherwise.